pkg/restore/client.go

```diff
@@ -619,89 +730,97 @@ func (rc *Client) switchTiKVMode(ctx context.Context, mode import_sstpb.SwitchMo
 	return nil
 }
 
-// ValidateChecksum validate checksum after restore.
-func (rc *Client) ValidateChecksum(
+// GoValidateChecksum forks a goroutine to validate checksum after restore.
+// it returns a channel fires a struct{} when all things get done.
+func (rc *Client) GoValidateChecksum(
 	ctx context.Context,
+	tableStream <-chan CreatedTable,
 	kvClient kv.Client,
-	tables []*utils.Table,
-	newTables []*model.TableInfo,
+	errCh chan<- error,
 	updateCh glue.Progress,
-) error {
+) <-chan struct{} {
 	start := time.Now()
 	defer func() {
 		elapsed := time.Since(start)
 		summary.CollectDuration("restore checksum", elapsed)
 	}()
```

Comment:
the timer (last 4 lines) no longer make any sense now. move them into the goroutine please.